docs/metadata.md

```diff
@@ -105,6 +105,32 @@ To find out more about config command, have a look at the [config section]
 [conf].
 
 
+## Fixed assets
+
+Some metadata is provided in the form of conventions, such as license files,
+icons and desktop files among others. For these fixed files to make it into
+your final snap they need to be in a `setup` directory at the same level of
+your `snapcraft.yaml`.
+
+### Snap icon
+
+To provide an icon for your snap, while for command line applications might
+not seem as important it is rather important for discoverability from
+management interfaces such as store fronts like webdm.
```

Comment:
This could flow a bit better. Perhaps something like "Providing an icon for your snap is important, even for command-line applications, if for nothing else than discoverability from management[...]"